app/models/katello/upstream_consumer.rb

```diff
@@ -0,0 +1,61 @@
+require 'katello/resources/candlepin'
+
+module Katello
+  class UpstreamConsumer < OpenStruct
+    CP_CONSUMER = Resources::Candlepin::UpstreamConsumer
```

Comment:
Let's imagine we wanted to change the resource, and so we would modify the value of CP_CONSUMER. Something about that doesn't feel right. In my work to delete entitlements I've used a class method whose return value is Resources::Candlepin::UpstreamConsumer - so we can change the implementation of that method to achieve the desired behavior. Mind changing that if you agree? Or I can manage it when I rebase off of this!

Reply:
I don't really understand your point here. It's a constant, and thus intended to never change (indeed, I think Ruby might give a warning if you try). The CP resource for consumer is never going to change.

Reply:
'The CP resource for consumer is never going to change' - I don't think we can guarantee that. What if we started to live in a world where there were multiple upstreams? We'd be a bit closer to injecting that dependency by using a class method is my point - rather than modifying the value of a constant or omitting it for another approach altogether

Reply:
There already can be multiple upstreams (one for each org), and this will work just fine because each upstream Candlepin has the same consumer resource, regardless where it's running (katello, RHSM portal, staging, etc.). I'm still not really seeing your point (sorry?). It's a constant which is an alias for another constant, just like we did in the [UpstreamPool model](https://github.com/Katello/katello/blob/master/app/models/katello/upstream_pool.rb#L5).